src/main/java/graphql/schema/idl/TypeDefinitionRegistry.java

```diff
@@ -623,9 +623,25 @@ public boolean isPossibleType(Type abstractType, Type possibleType) {
             return false;
         } else {
             InterfaceTypeDefinition iFace = (InterfaceTypeDefinition) abstractTypeDef;
-            List<ImplementingTypeDefinition> implementingTypeDefinitions = getAllImplementationsOf(iFace);
-            return implementingTypeDefinitions.stream()
-                    .anyMatch(od -> od.getName().equals(targetObjectTypeDef.getName()));
+
+            for (TypeDefinition<?> t : types.values()) {
```

Comment:
Probably worth investing in a cached reverse lookup at some point instead of looking through all the types everytime. Maybe on that new read only type registry.

Reply:
How do you see this reverse lookup working - simply a map of name -> type say  or somethings else?

ps the read only type registry PR is now merged

Reply:
I think i know now - I suspect you want something like

```

    private final Map<InterfaceTypeDefinition, List<ImplementingTypeDefinition>> allImplementationsOf;
    private final Map<InterfaceTypeDefinition, List<ObjectTypeDefinition>> implementationsOf;


    @Override
    public List<ImplementingTypeDefinition> getAllImplementationsOf(InterfaceTypeDefinition targetInterface) {
        return allImplementationsOf.getOrDefault(targetInterface, ImmutableList.of());
    }

    @Override
    public List<ObjectTypeDefinition> getImplementationsOf(InterfaceTypeDefinition targetInterface) {
        return implementationsOf.getOrDefault(targetInterface, ImmutableList.of());
    }
```